### PR TITLE
Update to latest gt4py

### PR DESCRIPTION
**Pull request by jdahm**
## Purpose

*Work in progress*. Update to the latest gt4py version.

### CODE REVIEW

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

File: fv3core/docker/Dockerfile [hunk 1]
```diff
@@ -58,3 +58,5 @@ RUN chmod +x /entrypoint.sh && \
 ARG ENV_CUDA_PATH=""
 ENV CUDA_PATH=${ENV_CUDA_PATH}
 ENV IN_DOCKER=True
+ENV GT_CACHE_ROOT=/
+ENV GT_CACHE_DIR_NAME=gt_cache
```
Comment (jdahm) [inline fv3core/docker/Dockerfile:62]: Set as an ENV instead of inside the Dockerfile.

File: pace-util/tests/mpi/test_mpi_mock.py [hunk 2]
```diff
@@ -301,7 +301,7 @@ def test_worker(comm, dummy_results, mpi_results, numpy):
         assert len(dummy_results) == len(mpi_results)
         for dummy, mpi in zip(dummy_results, mpi_results):
             if isinstance(mpi, numpy.ndarray):
-                numpy.testing.assert_array_equal(dummy, mpi)
+                numpy.testing.assert_array_equal(dummy.data, mpi.data)
```
Comment (jdahm) [inline pace-util/tests/mpi/test_mpi_mock.py:304]: Double check that we can use `numpy.testing.assert_array_equal ` on `gt4py.storage.Storage` instances.

Comment (jdahm) [reply pace-util/tests/mpi/test_mpi_mock.py:304]: @mcgibbon Yes, this is supported. Can reproduce with
```python
import numpy as np
from gt4py import storage
default_origin = (0, 0, 0)
shape = (6, 6, 3)
for backend in ("gtc:numpy", "gtc:gt:cpu_ifirst"):
    args = (backend, default_origin, shape, float)
    np.testing.assert_array_equal(storage.ones(*args), storage.ones(*args))
```

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
To dos:
- [ ] Increase the test timeout for `gtc:numpy` on the c12_54ranks_standard` test set.
- [ ] Determine and resolve issue with `gtc:gt:gpu` on `PE_Halo`.

**Comment by rheacangeo** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Review by elynnwu — APPROVED** [review]

**Pull request merged**